### PR TITLE
Updated Keypair

### PR DESCRIPTION
**Pull request by JFGHT**
https://stellar.github.io/js-stellar-sdk/Keypair.html#rawPublicKey
https://github.com/stellar/js-stellar-base/blob/master/src/generated/stellar-xdr_generated.js

### CODE REVIEW

File: types/stellar-sdk/index.d.ts [hunk 1]
```diff
@@ -876,4 +879,9 @@ export namespace xdr {
     class Asset extends XDRStruct { }
     class Memo extends XDRStruct { }
     class TransactionEnvelope extends XDRStruct { }
+    class DecoratedSignature extends XDRStruct {
+      constructor(keys: { hint: SignatureHint, signature: Signature })
```
Comment (Carl-Foster) [inline types/stellar-sdk/index.d.ts]: Could you add the types:
`hint(): Buffer`
`signature(): Buffer`

I think...

Comment (JFGHT) [reply types/stellar-sdk/index.d.ts]: What do you mean?

Comment (Carl-Foster) [reply types/stellar-sdk/index.d.ts]: Those methods at least used to exist (I work on a fork). Those two functions on the DecoratedSignature can be used to do something like this to see who has signed a transaction envelope:
```js
  const transactionSignatures = transaction.signatures.map((sig: any) => sig.signature())

  return signers.filter(kp => transactionSignatures.some(sig => kp.verify(transaction.hash(), sig)))
```

Comment (JFGHT) [reply types/stellar-sdk/index.d.ts]: I just reviewed the source code files again and still didn't find those. I can add them if you want tho.

```
   struct DecoratedSignature
   {
       SignatureHint hint;  // last 4 bytes of the public key, used as a hint
       Signature signature; // actual signature
   };
```

**Comment by pselden** [comment]
@JFGHT -- I use them here: https://github.com/stellarguard/stellarguard/blob/c3c40b0b11500dd87e5b2e939727000e8de1e526/src/server/lib/stellar/signer.js#L25

They definitely exist and are of type Buffer.

**Comment by andy-ms** [comment]
@pselden PRs accepted.

**Comment by JFGHT** [comment]
@pselden Hello there!

I didn't find anything related to them in the source code. Feel free to include them in another PR.

File: types/stellar-sdk/index.d.ts (hunk 1 above)
Comment (Carl-Foster) [reply types/stellar-sdk/index.d.ts]: That is the source code that defines the functions I described above. XDR -> JS turns them into getters. Disappointing to not have added it in when you were just there.

**Pull request merged**